src/Record.js

```diff
@@ -78,6 +78,12 @@ export class Record extends KeyedCollection {
     if (!this.has(k)) {
       throw new Error('Cannot set unknown key "' + k + '" on ' + recordName(this));
     }
+    if (this._map && !this._map.has(k)) {
```

Comment:
should this be `!this._map || !this._map.has(k)` to also include the case where there is no map yet?